src/modules/directory.rs

```diff
@@ -64,33 +63,33 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
     // Truncate the dir string to the maximum number of path components
     let truncated_dir_string = truncate(dir_string, config.truncation_length as usize);
 
-    if config.fish_style_pwd_dir_length > 0 {
+    let fish_prefix = if config.fish_style_pwd_dir_length > 0 {
         // If user is using fish style path, we need to add the segment first
         let contracted_home_dir = contract_path(&current_dir, &home_dir, HOME_SYMBOL);
-        let fish_style_dir = to_fish_style(
+        to_fish_style(
             config.fish_style_pwd_dir_length as usize,
             contracted_home_dir,
             &truncated_dir_string,
-        );
-
-        module.create_segment(
-            "path",
-            &SegmentConfig {
-                value: &fish_style_dir,
-                style: None,
-            },
-        );
-    }
+        )
+    } else {
+        String::from("")
+    };
 
-    module.create_segment(
-        "path",
-        &SegmentConfig {
-            value: &truncated_dir_string,
-            style: None,
-        },
-    );
+    let final_dir_string = format!("{}{}", fish_prefix, truncated_dir_string);
```

Comment:
I did test with the `fish_style_pwd_dir_length` configuration value > 0, and that seemed to work when comparing to the official build.